tronweb2/src/components/ActionGraph/ActionGraph.js

```diff
@@ -79,21 +72,62 @@ function extendActionData(actionList) {
   return [actionMap, numCols, numLevels];
 }
 
-function buildGraph(actionData, height, width) {
+function shortenName(name) {
+  const maxTextLength = 20;
+  if (name.length > maxTextLength) {
+    return `${name.substring(0, maxTextLength)}...`;
+  }
+  return name;
+}
+
+function actionTooltip(action) {
+  return (
+    <Tooltip>
+      <div className="tooltip-action-name p-2">
+        {action.name}
+      </div>
+      <div className="p-2">
+        command:
+        {' '}
+        {action.command}
+      </div>
+    </Tooltip>
+  );
+}
+
+function buildGraph(actionData, minWidth) {
   const [actionMap, numCols, numLevels] = actionData;
   const nodeList = Object.values(actionMap);
+
+  const actualWidth = Math.max(minWidth, numCols * 80);
+  const actualHeight = (numLevels + 1) * 100;
+
+  const rectHeight = 36;
+  const rectSidePadding = 10;
+
   const nodeYScale = scaleLinear()
     .domain([0, numLevels])
-    .range([0, height - 30]);
+    .range([0, actualHeight - rectHeight - 5]);
   const nodeXScale = scaleLinear()
     .domain([0, numCols])
-    .range([0, width]);
+    .range([0, actualWidth]);
 
   const nodeElements = nodeList.map((d) => (
-    <OverlayTrigger key={d.name} placement="top" overlay={<Tooltip><div>{d.name}</div></Tooltip>}>
+    <OverlayTrigger key={d.name} placement="left" overlay={actionTooltip(d)}>
       <g className="node">
-        <circle r="6" cx={nodeXScale(d.col)} cy={nodeYScale(d.level)} />
-        <text dx={nodeXScale(d.col) + 10} dy={nodeYScale(d.level)} className={d.type}>
+        <rect
+          x={nodeXScale(d.col) - rectSidePadding}
+          y={nodeYScale(d.level) - rectHeight / 2}
+          rx="5"
+          ry="5"
+          width={(shortenName(d.name).length * 7) + (2 * rectSidePadding) + 1}
+          height={rectHeight}
+          stroke="black"
```

Comment:
I feel like the black is a bit harsh here.  Could we lighten it a little bit, or use a drop-shadow instead?

I also don't care that much so feel free to drop.

Reply:
I'd like a drop shadow too, but it's pretty complex to define in svg, so I don't want to spend too much time getting it right for now.